braindecode/models/eegitnet.py

```diff
@@ -0,0 +1,234 @@
+    def __init__(self, in_ch, kernel_sz, dialation, padding, drop_prob=0.4):
+        super().__init__()
+        self.pad=padding
+        self.tc1 = nn.Sequential(
+            *[
+                _DepthwiseConv2d(
+                    in_ch,
+                    kernel_size=(1, kernel_sz),
+                    depth_multiplier=1,
+                    dilation=(1, dialation),
+                    bias=False,
+                    padding="valid",
+                ),
+                nn.BatchNorm2d(in_ch),
+                nn.ELU(),
+                nn.Dropout(drop_prob),
+            ]
+        )
+
+        self.tc2 = nn.Sequential(
+            *[
+                _DepthwiseConv2d(
+                    in_ch,
+                    kernel_size=(1, kernel_sz),
+                    depth_multiplier=1,
+                    dilation=(1, dialation),
+                    bias=False,
+                    padding="valid",
+                ),
+                nn.BatchNorm2d(in_ch),
+                nn.ELU(),
+                nn.Dropout(drop_prob),
+            ]
+        )
+
+    def forward(self, x):
+        paddings = (self.pad, 0, 0, 0, 0, 0, 0, 0)
+        x = nn.functional.pad(x, paddings)
+        x = self.tc1(x)
+        x = nn.functional.pad(x, paddings)
+        x = self.tc2(x)
+        return x
+
+
+class EEG_ITNET(nn.Sequential):
+    """ EEG-ITNet: An Explainable Inception Temporal
+     Convolutional Network for motor imagery classification from
+     Salami et. al 2022.
+
+    See [Salami2022]_ for details.
+
+    Code adapted from https://github.com/abbassalami/eeg-itnet
+
+    Parameters
+    ----------
+    n_classes: int
+        number of outputs of the decoding task (for example number of classes in
+        classification)
+    n_in_chans: int
+        number of input EEG channels
+    input_window_samples : int
+        Number of time samples.
+    drop_prob: float
+        Dropout probability.
+
+    References
+    ----------
+    .. [Salami2022] A. Salami, J. Andreu-Perez and H. Gillmeister, "EEG-ITNet: An Explainable
+    Inception Temporal Convolutional Network for motor imagery classification," in IEEE Access,
+    doi: 10.1109/ACCESS.2022.3161489.
+    
+    Notes
+    -----
+    This implementation is not guaranteed to be correct, has not been checked
+    by original authors, only reimplemented from the paper based on author implementation.
+    """
+
+    def __init__(self, n_classes, in_channels, input_window_samples,
+                 drop_prob=0.4):
+        super().__init__()
+
+        
+        # ======== handling EEG input ========================
+        def _permute(x):
+            """
+            Permutes data:
+
+            from dim:
+            batch, chans, time, 1
+
+            to dim:
+            batch, 1, chans, time
+
+            """
+            return x.permute([0, 3, 1, 2])
+
+        self.prep_input = nn.Sequential(
+            Ensure4d(),
+            Expression(_permute))
```

Comment:
The idea here with `nn,Sequential` would be to use `add_module` method instead of creating class attributes, for example:
```suggestion
        self.add_module(nn.Sequential(
            Ensure4d(),
            Expression(_permute)
        )
```